index.js

```diff
@@ -31,13 +31,85 @@ SOFTWARE.
 var parseXML     = require('xml2js').parseString
   , url          = require('url')
   , EventEmitter = require('events').EventEmitter
+  , HttpProxyAgent = require('http-proxy-agent')
+  , HttpsProxyAgent = require('https-proxy-agent')
   ;
 
 // These numbers were obtained by measuring and averaging both using this module and the official speedtest.net
 var speedTestDownloadCorrectionFactor = 1.135
   , speedTestUploadCorrectionFactor   = 1.139
   ;
 
+// 	Proxy by parameter
+var proxyOptions = null;
+
+// Search a proxy (http|https) in env with case-insensitive
+var proxyHttpEnv = findPropertiesInEnvInsensitive("HTTP_PROXY");
+var proxyHttpsEnv = findPropertiesInEnvInsensitive("HTTPS_PROXY");
+
+function findPropertiesInEnvInsensitive(prop) {
+    prop = (prop + "").toLowerCase();
+    for(var p in process.env){
+        if(process.env.hasOwnProperty(p) && prop == (p+ "").toLowerCase()){
```

Comment:
`p` is already guaranteed to be a string, so it's not necessary to add the `+ ""`.

Reply:
done